org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/TextEditConverter.java

```diff
@@ -78,9 +92,12 @@ public TextDocumentEdit convertToTextDocumentEdit(int version) {
 	@Override
 	public boolean visit(InsertEdit edit) {
 		try {
-			org.eclipse.lsp4j.TextEdit te = new org.eclipse.lsp4j.TextEdit();
+			org.eclipse.lsp4j.TextEdit te = this.changeAnnotation != null ? new org.eclipse.lsp4j.AnnotatedTextEdit() : new org.eclipse.lsp4j.TextEdit();
```

Comment:
You can extract the text edit creation to a method, in that method, set the annotation id if `this.changeAnnotation != null`. - avoid calling `instanceof` in each `visit()s`.

Reply:
fixed in https://github.com/eclipse/eclipse.jdt.ls/pull/2497/commits/e6bd4c270ebcd408aceae68713eed712c5f88879